validphys2/src/validphys/commondataparser.py

```diff
@@ -40,21 +108,805 @@
+
+    def get_label(self, var):
+        """For the given variable, return the label as label (unit)
+        If the label is an "extra" return the last one
+        """
+        if var.startswith("extra_"):
+            return list(self.variables.values())[-1]
+        return self.variables[var].full_label()
+
+    def apply_label(self, var, value):
+        """For a given value for a given variable, return the labels
+        as label = value (unit)
+        If the variable is not include in the list of variables, returns None
+        as the variable could've been transformed by a kinematic transformation
+        """
+        if var not in self.variables:
+            return None
+        return self.variables[var].apply_label(value)
+
+
+###
+
+
+### Observable and dataset definitions
+@dataclasses.dataclass(frozen=True)
+class ObservableMetaData:
+    observable_name: str
+    observable: dict
+    ndata: int
+    # Plotting options
+    plotting: PlottingOptions
+    process_type: str
+    kinematic_coverage: list[str]
+
+    # Data itself
+    kinematics: ValidKinematics
+    data_uncertainties: list[ValidPath]
+
+    # The central data is optional _only_ for
+    # positivity datasets, and will be checked as soon as the class is instantiated
+    data_central: Optional[ValidPath] = None
+
+    # Optional data
+    theory: Optional[TheoryMeta] = None
+    tables: Optional[list] = dataclasses.field(default_factory=list)
+    npoints: Optional[list] = dataclasses.field(default_factory=list)
+    variants: Optional[ValidVariants] = dataclasses.field(default_factory=dict)
+    applied_variant: Optional[str] = None
+    ported_from: Optional[str] = None
+
+    # Derived quantities:
+    # Note that an observable without a parent will fail in many different ways
+    _parent: Optional[Any] = None
+
+    def __post_init__(self):
+        """
+        Small modifications for better compatibility with the rest of validphys
+        """
+        # Since vp will rely on the kinematics being 3 variables,
+        # fill the extra with whatever can be found in the kinematics dictionary
+        # otherwise just fill with extra_x
+        if len(self.kinematic_coverage) < 3:
+            unused = list(set(self.kinematics.variables) - set(self.kinematic_coverage))
+            diff_to_3 = 3 - len(self.kinematic_coverage)
+            if unused:
+                self.kinematic_coverage += unused[diff_to_3:]
+            else:
+                self.kinematic_coverage += [f"extra_{i}" for i in range(diff_to_3)]
+
+        object.__setattr__(self, 'process_type', self.process_type.upper())
+
+    def check(self):
+        """Various checks to apply manually to the observable before it is used anywhere
+        These are not part of the __post_init__ call since they can only happen after the metadata
+        has been read and the observable selected.
+        """
+        # Check that the data_central is empty if and only if the dataset is a positivity set
```

Comment:
This also applies to integrability dataset, not only to positivity.
```suggestion
        # Check that the data_central is empty if and only if the dataset is a positivity/integrability set
```